test/e2e/framework/pv_util.go

```diff
@@ -688,3 +692,40 @@ func deletePD(pdName string) error {
 		return fmt.Errorf("Provider does not support volume deletion")
 	}
 }
+
+func CreatePVSource(zone string) (*v1.PersistentVolumeSource, error) {
+	diskName, err := CreatePDWithRetry(zone)
+	if err != nil {
+		return nil, err
+	}
+
+	if TestContext.Provider == "gce" || TestContext.Provider == "gke" {
+		return &v1.PersistentVolumeSource{
+			GCEPersistentDisk: &v1.GCEPersistentDiskVolumeSource{
+				PDName:   diskName,
+				FSType:   "ext3",
+				ReadOnly: false,
+			},
+		}, nil
+	} else if TestContext.Provider == "aws" {
+		return &v1.PersistentVolumeSource{
+			AWSElasticBlockStore: &v1.AWSElasticBlockStoreVolumeSource{
+				VolumeID: diskName,
```

Comment:
I don't know if this is correct.  I don't have an AWS environment to test with.  Currently, this codepath is not called for AWS.